lib/Conch/Route/Device.pm

```diff
@@ -14,128 +14,496 @@ Conch::Route::Device
 
 Sets up the routes for /device:
 
-    GET     /device/?hostname=:host, ?mac=:mac, ?ipaddr=:ipaddr, ?:setting_key=:setting_value
-    GET     /device/:device_id
-    GET     /device/:device_id/pxe
-    GET     /device/:device_id/phase
-    POST    /device/:device_id
-    POST    /device/:device_id/graduate
-    POST    /device/:device_id/triton_setup
-    POST    /device/:device_id/triton_uuid
-    POST    /device/:device_id/triton_reboot
-    POST    /device/:device_id/asset_tag
-    POST    /device/:device_id/validated
-    POST    /device/:device_id/phase
-
-    GET     /device/:device_id/location
-    POST    /device/:device_id/location
-    DELETE  /device/:device_id/location
-
-    GET     /device/:device_id/settings
-    POST    /device/:device_id/settings
-    GET     /device/:device_id/settings/#key
-    POST    /device/:device_id/settings/#key
-    DELETE  /device/:device_id/settings/#key
-
-    POST    /device/:device_id/validation/:validation_id
-    POST    /device/:device_id/validation_plan/:validation_plan_id
-    GET     /device/:device_id/validation_state
-
-    GET     /device/:device_id/interface
-    GET     /device/:device_id/interface/:interface_name
-    GET     /device/:device_id/interface/:interface_name/:field
-
 =cut
 
 sub routes {
     my $class = shift;
     my $device = shift; # secured, under /device
 
+=head3 C<POST /device/:device_id>
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Response Schema: L<ValidationStateWithResults|https://conch.joyent.us/schema/response/ValidationStateWithResults>
+
+=back
+
+=cut
+
     # POST /device/:device_id
     $device->post('/:device_id')->to('device_report#process');
 
+=head3 C<GET /device?key=:value>
+
+Looks up one or more devices by query parameter. Supports:
+
+    /device?hostname=$hostname
+    /device?mac=$macaddr
+    /device?ipaddr=$ipaddr
+    /device?$setting_key=$setting_value
```

Comment:
What is this setting option? We know it's a device setting but that's not clear from the text.

Reply:
I've expanded the explanation of this setting in the copy.